src/metrics/scrapper.py

```diff
@@ -1,32 +1,125 @@
 # Python Imports
 import requests
-from typing import List
 import logging
+import pandas as pd
+from itertools import chain
+from typing import List, Dict
+from requests import Response
+from pathlib import Path
 
 # Project Imports
-import src.logging.logger
 from src.metrics import scrape_utils
+from result import Ok, Err, Result
 
 logger = logging.getLogger(__name__)
 
 
 class Scrapper:
-    def __init__(self, url: str, namespace: str, metrics: List):
+    def __init__(self, url: str, namespace: str, out_folder: str, metrics: List):
         self._url = url
         self._namespace = namespace
+        self._out_folder = out_folder
         self._metrics = metrics
         # TODO make interval match value in cluster
-        self._template = "irate($metric{namespace=$namespace}[3m])"
+        self._template = 'irate($metric{namespace=$namespace}[3m])'
 
-    def make_queries(self):
+    def query_and_dump_metrics(self):
         for metric in self._metrics:
-            query = self._template.replace("$metric", metric)
-            query = query.replace("$namespace", self._namespace)
-            promql = scrape_utils.create_promql(self._url, query, 1, 60)
-            logger.info(f"Promql: {promql}")
-            response = requests.get(promql)
-            logger.info(f"Response: {response.status_code}")
+            logger.info(f'Querying {metric}')
+            promql = self._create_query(metric)
 
+            result = self._make_query(promql)
+            if result.is_err():
+                logger.warning(f'Error querying {metric}. {result.err_value}')
 
+            response = result.ok_value
```

Comment:
You're not exiting the function after the `is_err` check, which means that if there is an error you'll trigger an exception (if that's how the underlying library handles it) when accessing `result.ok_value`.

Reply:
Yup, the idea was to add a `continue` to go to the next metric. Thanks.